src/Roslyn.Suppressions/Roslyn.Suppressions.UnitTests/SuppressedIssuesFileStorageTests.cs

```diff
@@ -0,0 +1,258 @@
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.Suppression
+{
+    [TestClass]
+    public class SuppressedIssuesFileStorageTests
+    {
+
+        [TestMethod]
+        public void Update_Should_WriteIssuesToFilePath()
+        {
+            SonarQubeIssue issue1 = CreateIssue("issueKey1");
+            SonarQubeIssue issue2 = CreateIssue("issueKey2");
+
+            var issues = new List<SonarQubeIssue> { issue1, issue2 };
+
+            var logger = new TestLogger();
+
+            var directory = new Mock<IDirectory>();
+            var file = new Mock<IFile>();
+            Mock<IFileSystem> fileSystem = CreateFileSystem(directory, file);
+
+            var fileStorage = new SuppressedIssuesFileStorage(logger, fileSystem.Object);
+
+            fileStorage.Update("projectKey", issues);
+
+            directory.Verify(d => d.CreateDirectory(GetTempPath()), Times.Once);
+            file.Verify(f => f.WriteAllText(GetFilePath("projectKey"), JsonConvert.SerializeObject(issues)), Times.Once);
+            logger.AssertNoOutputMessages();
+        }
+
+        [TestMethod]
+        public void Get_Should_ReadIssuesFromFilePath()
+        {
+            SonarQubeIssue issue1 = CreateIssue("issueKey1");
+            SonarQubeIssue issue2 = CreateIssue("issueKey2");
+            var issues = new List<SonarQubeIssue> { issue1, issue2 };
+            
+            var logger = new TestLogger();
+
+            var directory = new Mock<IDirectory>();
+            var file = CreateFileForGet("projectKey", issues);
+            Mock<IFileSystem> fileSystem = CreateFileSystem(directory, file);
+
+            var fileStorage = new SuppressedIssuesFileStorage(logger, fileSystem.Object);
+
+            var issuesGotten = fileStorage.Get("projectKey").ToList();
+
+            directory.Verify(d => d.CreateDirectory(It.IsAny<string>()), Times.Never);
+            file.Verify(f => f.ReadAllText(GetFilePath("projectKey")), Times.Once);
+            logger.AssertNoOutputMessages();
+
+            Assert.AreEqual(2, issuesGotten?.Count);
+            Assert.AreEqual(issue1.IssueKey, issuesGotten[0].IssueKey);
+            Assert.AreEqual(issue2.IssueKey, issuesGotten[1].IssueKey);
+        }
+
+        [DataRow(null)]
+        [DataRow("")]
+        [DataRow("  ")]
+        [TestMethod]
+        public void Update_Should_ThrowWhenProjectKeyIsNull(string projectKey)
+        {
+            SonarQubeIssue issue = CreateIssue("issueKey");
+            var issues = new List<SonarQubeIssue> { issue };
+
+            var logger = new TestLogger();
+
+            var directory = new Mock<IDirectory>();
+            var file = new Mock<IFile>();
+            Mock<IFileSystem> fileSystem = CreateFileSystem(directory, file);
+
+            var fileStorage = new SuppressedIssuesFileStorage(logger, fileSystem.Object);
+            var ex = Assert.ThrowsException<ArgumentException>(() => fileStorage.Update(projectKey, issues));
```

Comment:
Another minor team convention: we use FluentAssertions so it would be written instead 
```
Action act = () => fileStorage.Update(projectKey, issues);
act.Should().ThrowExactly<ArgumentException>().And.ParamName.Should().Be....
```